pkg/runtime/options.go

```diff
@@ -4,23 +4,23 @@ import (
 	"github.com/dapr/dapr/pkg/components/bindings"
 	"github.com/dapr/dapr/pkg/components/exporters"
 	"github.com/dapr/dapr/pkg/components/middleware/http"
+	"github.com/dapr/dapr/pkg/components/nameresolution"
 	"github.com/dapr/dapr/pkg/components/pubsub"
 	"github.com/dapr/dapr/pkg/components/secretstores"
-	"github.com/dapr/dapr/pkg/components/servicediscovery"
 	"github.com/dapr/dapr/pkg/components/state"
 )
 
 type (
 	// runtimeOpts encapsulates the components to include in the runtime.
 	runtimeOpts struct {
-		secretStores     []secretstores.SecretStore
-		states           []state.State
-		pubsubs          []pubsub.PubSub
-		exporters        []exporters.Exporter
-		serviceDiscovery []servicediscovery.ServiceDiscovery
-		inputBindings    []bindings.InputBinding
-		outputBindings   []bindings.OutputBinding
-		httpMiddleware   []http.Middleware
+		secretStores    []secretstores.SecretStore
+		states          []state.State
+		pubsubs         []pubsub.PubSub
+		exporters       []exporters.Exporter
+		nameResolutions []nameresolution.NameResolution
```

Comment:
nameResolvers?

Reply:
I would follow the existing convention to be consistent with the existing ones - `[component type]s`